documentation/src/main/asciidoc/user_guide/server_protocols_hotrod_transaction.adoc

```diff
@@ -55,94 +49,101 @@ cb.transaction().timeout(1, TimeUnit.MINUTES)
+`getCache(String cacheName, boolean forceReturnValue, TransactionManager transactionManager)`::
+Same as previous, but can also force return values for write operations.
 
-* `getCache(String cacheName, TransactionMode transactionMode, TransactionManager transactionManager)`:
-it returns a {rc} overriding the configured {tm} and {tx-mode}.
-If `transactionManager` or `transactionMode` is null, it will use the configured values.
+`getCache(String cacheName, TransactionMode transactionMode, TransactionManager transactionManager)`::
+Returns a {rc} and overrides the configured {tm} and {tx-mode}. Uses the configured values, if `transactionManager` or `transactionMode` is null.
 
-* `getCache(String cacheName, boolean forceReturnValue, TransactionMode transactionMode, TransactionManager transactionManager)`:
-same as above but forcing (or not) a return value for write operations.
+`getCache(String cacheName, boolean forceReturnValue, TransactionMode transactionMode, TransactionManager transactionManager)`::
+Same as previous, but can also force return values for write operations.
 
-The {rc} was extended with `getTransactionManager()` to return the {tm} used by the cache.
-It is `null` if the {rc} isn't transactional.
+[NOTE]
+====
+The `getCache(String)` method returns {rc} instances regardless of whether they are transaction or not. {rc} includes a `getTransactionManager()` method that returns the {tm} that the cache uses. If the {rc} is not transactional, the method returns `null`.
+====
 
-===== Effect of `forceReturnValue` in transactions
+[[hr_transactions_force_return_value]]
+===== Detecting Conflicts with Transactions
+Transactions use the initial values of keys to detect conflicts. For example, "k" has a value of "v" when a transaction begins. During the prepare phase, the transaction fetches "k" from the server to read the value. If the value has changed, the transaction rolls back to avoid a conflict.
 
-The `forceReturnValue` parameter affects what write operations return when a key is accessed for the first time.
-If set to `false`, the client optimizes and it doesn't fetch the current value from the server.
-Otherwise, the client fetches the most up-to-date value and performs the write operation.
+[NOTE]
+====
+Transactions use versions to detect changes instead of checking value equality.
+====
 
-NOTE: It doesn't affect "conditional" writes (such as `replace` or `putIfAbsent`) since they need the most up-to-date
-value to perform properly.
+The `forceReturnValue` parameter controls write operations to the {rc} and helps avoid conflicts. It has the following values:
 
-Each transaction keeps track of the values read in order to detect conflicts.
-As an example, if a transaction reads value `v` and, at prepare time, the value is `v1`, the transaction is rolled back.
+* If `true`, the {tm} fetches the most recent value from the server before performing write operations.
```

Comment:
do you think it would be nice to say that `force-return-value` only affects writes that access the key for the first time?